portal/v2/src/ClusterDetailList.tsx

```diff
@@ -38,38 +41,85 @@ interface IClusterDetailComponentState {
   detailPanelSelected: string
 }
 
-export class ClusterDetailComponent extends Component<ClusterDetailComponentProps, IClusterDetailComponentState> {
-
+export class ClusterDetailComponent extends Component<
+  ClusterDetailComponentProps,
+  IClusterDetailComponentState
+> {
   constructor(props: ClusterDetailComponentProps | Readonly<ClusterDetailComponentProps>) {
-    super(props);
+    super(props)
   }
 
   public render() {
-    switch (this.props.detailPanelVisible.toLowerCase()) {
-      case "overview":
-      {
-        return (
-          <OverviewWrapper clusterName= {this.props.item.name} currentCluster={this.props.cluster!} detailPanelSelected={this.props.detailPanelVisible} loaded={this.props.isDataLoaded}/>
-        )
-      }
-      case "nodes":
-        {
-          return (
-            <NodesWrapper currentCluster={this.props.cluster!} detailPanelSelected={this.props.detailPanelVisible} loaded={this.props.isDataLoaded}/>
-          );
-        }
-        case "machines":
-        {
-          return (
-            <MachinesWrapper currentCluster={this.props.cluster!} detailPanelSelected={this.props.detailPanelVisible} loaded={this.props.isDataLoaded}/>
-          );
-        }
-        case "machinesets":
-        {
-          return (
-            <MachineSetsWrapper currentCluster={this.props.cluster!} detailPanelSelected={this.props.detailPanelVisible} loaded={this.props.isDataLoaded}/>
-          );
-        }
-      }
+    interface Map {
+      [key: string]: JSX.Element
+    }
+
+    const menus: Map = {
```

Comment:
I think Daniel meant more like this https://github.com/Azure/ARO-RP/blob/9cb41357eb32f70a1f872be37435c3baeeb084fc/portal/v2/src/ClusterDetailList.tsx 

If you revert this change we can just handle it in the merge conflict